src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAddSolutionItemService.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Concurrent;
+using System.Composition;
+using System.IO;
+using System.Threading;
+using EnvDTE;
+using EnvDTE80;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Roslyn.Utilities;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    [ExportWorkspaceService(typeof(IAddSolutionItemService)), Shared]
+    internal partial class VisualStudioAddSolutionItemService : IAddSolutionItemService
+    {
+        private const string SolutionItemsFolderName = "Solution Items";
+
+        private readonly object _gate = new object();
+        private readonly IThreadingContext _threadingContext;
+        private readonly DTE _dte;
+        private readonly IVsFileChangeEx _fileChangeService;
+        private readonly ConcurrentDictionary<string, FileChangeTracker> _fileChangeTrackers;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public VisualStudioAddSolutionItemService(
+            IThreadingContext threadingContext,
+            SVsServiceProvider serviceProvider)
+        {
+            _threadingContext = threadingContext;
+            _dte = (DTE)serviceProvider.GetService(typeof(DTE));
+            _fileChangeService = (IVsFileChangeEx)serviceProvider.GetService(typeof(SVsFileChangeEx));
```

Comment:
❗️ MEF constructors must not have threading dependencies

Reply:
Thanks, fixed.